### PR TITLE
Feature/ex 5225 add smoke tests mobile

### PR DESCRIPTION
**Pull request by mavmaf**
## Motivation and context
The aim of this PR is to adapt the tests created in https://searchbroker.atlassian.net/browse/EX-5172 in order to run them as well for mobile view. For this purpose, `cy.viewport` was used, which offers a group of preset options. `macbook-13` and `iphone-7` were chosen.

As already commented on , there are still pending tests to be added in a different task (infinite scroll, URL navigation, extra params...).

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that causes existing functionality to not work as expected)
- [ ] Change requires a documentation update

## What is the destination branch of this PR?
- [x] `Main`
- [ ] Other. Specify:

## How has this been tested?
As usual, change line #10 in `cypress.json` to the following to run the test separately:
`  "testFiles": "**/*.feature",`

Tests performed according to [testing guidelines](https://github.com/empathyco/x/blob/main/.github/contributing/tests.md): 

## Checklist:

- [x] My code follows the **[style guidelines](https://github.com/empathyco/x/blob/main/.github/CONTRIBUTING.md#style-guides)** of this project.
- [x] I have performed a **self-review** on my own code.
- [ ] I have **commented** my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate **no new warnings**.
- [x] I have added **tests** that prove my fix is effective or that my feature works.
- [x] New and existing **unit tests pass locally** with my changes.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -1,3 +1,5 @@
 XComponentsPipeline {
     instance = 'Archetype'
+      build = true
```
Comment (tiborux) [inline Jenkinsfile:3]: You don't need this build parameter, remove it and format the file, please.

File: package.json [hunk 2]
```diff
@@ -9,8 +9,13 @@
   "scripts": {
     "serve": "vue-cli-service serve --open --skip-plugins eslint",
     "build": "rollup -c",
-    "test:e2e": "vue-cli-service test:e2e",
-    "test:e2e:ci": "vue-cli-service test:e2e --headless --browser chrome",
+    "build:serve": "concurrently \"rollup -c -w\" \"npm start\"",
```
Comment (tiborux) [inline package.json:12]:
```suggestion
    "build:serve": "concurrently \"rollup -c -w\" \"npm run serve:dist\"",
```

File: package.json [hunk 3]
```diff
@@ -54,26 +60,31 @@
     "@vue/test-utils": "~1.0.3",
     "autoprefixer": "~9.7.4",
     "commitizen": "~4.0.3",
+    "concurrently": "~6.5.1",
     "cssnano": "~4.1.10",
+    "cypress-cucumber-preprocessor": "^4.3.1",
```
Comment (tiborux) [inline package.json:65]: Use patch version

```suggestion
    "cypress-cucumber-preprocessor": "~4.3.1",
```

File: tests/e2e/cucumber/column-picker.feature [hunk 4]
```diff
@@ -1,14 +1,14 @@
...
     When  value <columns> from column picker list is clicked
     Then  column picker list value <columns> is selected
     And   grid shows <columns> results per row
     Examples:
-      | query   | columns |
-      | shirt   | 6       |
-      | shirt   | 0       |
+      | query   | columns | view        |
+      | shirt   | 6       | macbook-13  |
+      | shirt   | 0       | macbook-13  |
```
Comment (tiborux) [inline tests/e2e/cucumber/column-picker.feature:14]:
```suggestion
      | shirt   | 0       | iphone-7  |
```

File: tests/e2e/cucumber/facets.spec.ts [hunk 5]
```diff
@@ -47,9 +59,7 @@ When('clear filters button is clicked', () => {
 });
 
 // Scenario 4
-When('facet {string} is {string}', (facetName: string) => {
-  cy.getByDataTest(facetName).click();
-});
+
```
Comment (tiborux) [inline tests/e2e/cucumber/facets.spec.ts:62]: remove this white line

**Review by tiborux — CHANGES_REQUESTED** [review]
I changed the branched to point to the other one, just to see the new changes.

**Review by guillei10 — CHANGES_REQUESTED** [review]
review Beltran's comments and we are ready to go.

File: tests/e2e/cucumber/column-picker.feature (hunk 4 above)
Comment (mavmaf) [reply tests/e2e/cucumber/column-picker.feature]: There is no column-picker for mobile, so actually in this case there should be only one row:
```
    Examples:
      | query   | columns | view               |
      | shirt     | 6             | macbook-13  |
```

**Review by guillei10 — DISMISSED** [review]

**Comment by tiborux** [comment]
There is a problem with one test, do you have the same problem locally?

image

**Review by tiborux — APPROVED** [review]

**Review by guillei10 — APPROVED** [review]

**Pull request merged**